napalm_yang/parser.py

```diff
@@ -56,9 +56,15 @@ def _execute_methods(self, device, methods):
         result = []
         for m in methods:
             attr = device
+            args = []
+            kwargs = {}
             for p in m["method"].split("."):
                 attr = getattr(attr, p)
-            r = attr(**m["args"])
+            if isinstance(m.get("args", None), list):
```

Comment:
Ok, I see... I suggest doing this instead:

```
kwargs = m.get("kwargs", {})
if not isinstance(kwargs, dict):
    raise TypeError("kwargs has to be a dict")
```

Otherwise it seems like you are just ignoring invalid arguments without providing any feedback to the user.